arwen/elrondapi/manBufOps.go

```diff
@@ -0,0 +1,415 @@
+
+	gasToUse := metering.GasSchedule().ManagedBufferAPICost.MBufferSetBytes
+	metering.UseGas(gasToUse)
+	managedType.ConsumeGasForThisIntNumberOfBytes(int(dataLength))
+
+	data, err := runtime.MemLoad(dataOffset, dataLength)
+	if arwen.WithFault(err, context, runtime.ManagedBufferAPIErrorShouldFailExecution()) {
+		return -1
+	}
+
+	isSuccess := managedType.SetBytesForThisManagedBuffer(mBufferHandle, data)
+	if !isSuccess {
+		arwen.WithFault(arwen.ErrNoManagedBufferUnderThisHandle, context, runtime.ManagedBufferAPIErrorShouldFailExecution())
+		return 1
+	}
+
+	gasToUse = math.MulUint64(metering.GasSchedule().BaseOperationCost.DataCopyPerByte, uint64(len(data)))
+	metering.UseGas(gasToUse)
+
+	return 0
+}
+
+//export v1_4_mBufferGetLength
+func v1_4_mBufferGetLength(context unsafe.Pointer, mBufferHandle int32) int32 {
+	managedType := arwen.GetManagedTypesContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	gasToUse := metering.GasSchedule().ManagedBufferAPICost.MBufferGetLength
+	metering.UseGas(gasToUse)
+
+	length := managedType.GetLengthForThisManagedBuffer(mBufferHandle)
+	if length == -1 {
+		arwen.WithFault(arwen.ErrNoManagedBufferUnderThisHandle, context, runtime.ManagedBufferAPIErrorShouldFailExecution())
+		return -1
+	}
+
+	return length
+}
+
+//export v1_4_mBufferGetBytes
+func v1_4_mBufferGetBytes(context unsafe.Pointer, mBufferHandle int32, resultOffset int32) int32 {
+	managedType := arwen.GetManagedTypesContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	gasToUse := metering.GasSchedule().ManagedBufferAPICost.MBufferGetBytes
+	metering.UseGas(gasToUse)
+
+	managedBuffer, err := managedType.GetBytesForThisManagedBuffer(mBufferHandle)
+	if arwen.WithFault(err, context, runtime.ManagedBufferAPIErrorShouldFailExecution()) {
+		return 1
+	}
+	managedType.ConsumeGasForThisIntNumberOfBytes(len(managedBuffer))
+
+	err = runtime.MemStore(resultOffset, managedBuffer)
+	if arwen.WithFault(err, context, runtime.ManagedBufferAPIErrorShouldFailExecution()) {
+		return 1
+	}
+
+	gasToUse = math.MulUint64(metering.GasSchedule().BaseOperationCost.DataCopyPerByte, uint64(len(managedBuffer)))
+	metering.UseGas(gasToUse)
+
+	return 0
+}
+
+//export v1_4_mBufferExtendFromSlice
+func v1_4_mBufferExtendFromSlice(context unsafe.Pointer, mBufferHandle int32, dataOffset int32, dataLength int32) int32 {
+	managedType := arwen.GetManagedTypesContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	gasToUse := metering.GasSchedule().ManagedBufferAPICost.MBufferSetBytes
+	metering.UseGas(gasToUse)
+
+	data, err := runtime.MemLoad(dataOffset, dataLength)
+	if arwen.WithFault(err, context, runtime.ManagedBufferAPIErrorShouldFailExecution()) {
+		return 1
+	}
+
+	isSuccess := managedType.AppendBytesToThisManagedBuffer(mBufferHandle, data)
```

Comment:
The function name could be shorter for the managerType context - like a simple AppendBytes - there are other functions which are too long.

Reply:
there is no need to have in the function name "ToThisManagedBuffer"